### DISCUSSION TITLE
How to find and restore minimized applications?

### DISCUSSION THREAD

**Discussion by holocronweaver** [qtile/qtile#4660] [Q&A] [answered]
Assuming a default qtile config, if a window is minimized how can you find it to unminimize it?

For example, with qtile auto-minimize enabled, if I navigate away from a Steam game it is auto-minimized and I can hear it running in the background, but the window disappears. I cannot figure out how to restore the game as there is nowhere in the default qtile config or UI I can see to list minimized windows or to restore them.

**Comment by elParaguayo** [comment, marked as answer]
You could have a key bound to this:
```python
@lazy.group.function
def unminimize_all(group):
    for win in group.windows:
        if win.minimized:
            win.toggle_minimize()

keys = [
    Key([mod, "shift"], "m", unminimize_all, desc="Unminimize all windows in current group"),
    ...
]
```

**Reply by holocronweaver** [reply]
Thanks for the suggestion. ~~I tried this and it doesn't seem to unminimize Steam games - there is no visible effect.~~ (Got it working, though window is no longer fullscreen after being unminimized, and toggle_fullscreen seems to have no effect.) Is there a way to list minimized apps and unminimize individual apps?

**Reply by elParaguayo** [reply]
```python
from libqtile.command.client import InteractiveCommandClient

c = InteractiveCommandClient()

_, windows = c.group.eval("[w for w in self.windows if w.minimized]")

print(windows)
```
That will output something like
```
[Window(name='My Minimized Window', wid=85983235)]
```
You can now do:
```
qtile cmd-obj -o window 85983235 -f toggle_minimize
```
